src/figuresClasses.ts

```diff
@@ -1,11 +1,106 @@
-export interface Figure {}
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea: () => number;
+}
+
+export abstract class BasicFigure implements Figure {
+  shape: Shapes;
+
+  color: Colors;
+
+  constructor(shape: Shapes, color: Colors) {
+    this.color = color;
+    this.shape = shape;
+  }
+
+  abstract getArea(): number;
+}
+
+export class Triangle extends BasicFigure {
+  constructor(
+    color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super(Shapes.Triangle, color);
+
+    const maxValue: number = Math.max(this.a, this.b, this.c);
+    const sumOfOthersValues: number = this.a + this.b + this.c - maxValue;
+
+    if (
+      this.a <= 0 ||
+      this.b <= 0 ||
+      this.c <= 0 ||
+      maxValue >= sumOfOthersValues
+    ) {
+      throw new Error(
+        'Values must be > 0 and value of any cannot be more than sum of others',
+      );
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const halfPerimetr: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      halfPerimetr *
+        (halfPerimetr - this.a) *
+        (halfPerimetr - this.b) *
+        (halfPerimetr - this.c),
+    );
 
-export class Rectangle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle extends BasicFigure {
+  constructor(
+    color: Colors,
+    public radius: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - Similar to the `Triangle` class, you don't need to manually assign the `radius` parameter to a class property within the constructor when using an access modifier.

GOOD EXAMPLE:
constructor(color: Colors, public radius: number) {
  super(Shapes.Circle, color);
}